dio/lib/src/dio_event_processor.dart

```diff
@@ -0,0 +1,149 @@
+    // a chained exception. We can still add request information, so we do that.
+    if (dioError.stackTrace == null) {
+      return event.copyWith(
+        // Don't override just parts of the original request.
+        // Keep the original one or if there's none create one.
+        request: event.request ?? _toRequest(dioError),
+      );
+    }
+
+    try {
+      final exception = sentryExceptionFactory.getSentryException(
+        dioError.error,
+        stackTrace: dioError.stackTrace,
+      );
+
+      final exceptions = _removeDioErrorStackTraceFromValue(event, dioError);
+
+      return event.copyWith(
+        exceptions: [
+          exception,
+          ...exceptions,
+        ],
+        // Don't override just parts of the original request.
+        // Keep the original one or if there's none create one.
+        request: event.request ?? _toRequest(dioError),
+      );
+    } catch (e, stackTrace) {
+      _options.logger(
+        SentryLevel.debug,
+        'Could not convert DioError to SentryException',
+        exception: e,
+        stackTrace: stackTrace,
+      );
+    }
+    return event;
+  }
+
+  /// Remove the StackTrace from [dioError] so the message on Sentry looks
+  /// much better.
+  List<SentryException> _removeDioErrorStackTraceFromValue(
+    SentryEvent event,
+    DioError dioError,
+  ) {
+    // Don't edit the original list
+    final exceptions = List<SentryException>.from(
+      event.exceptions ?? <SentryException>[],
+    );
+
+    final dioErrorValue = dioError.toString();
+
+    final dioSentryExceptions =
+        exceptions.where((element) => element.value == dioErrorValue).toList();
+
+    if (dioSentryExceptions.isEmpty) {
+      return exceptions;
+    }
+    var e = dioSentryExceptions.first;
+    exceptions.removeWhere((element) => element == e);
+    dioError.stackTrace = null;
+    e = sentryExceptionFactory.getSentryException(dioError).copyWith(
+          mechanism: e.mechanism,
+          module: e.module,
+          stackTrace: e.stackTrace,
+          threadId: e.threadId,
+          type: e.type,
+        );
+
+    exceptions.add(e);
+
+    return exceptions;
+  }
+
+  SentryRequest? _toRequest(DioError dioError) {
+    final options = dioError.requestOptions;
+    // As far as I can tell there's no way to get the uri without the query part
+    // so we replace it with an empty string.
+    final urlWithoutQuery = options.uri.replace(query: '').toString();
+
+    final query = options.uri.query.isEmpty ? null : options.uri.query;
+
+    final headers = options.headers
+        .map((key, dynamic value) => MapEntry(key, value?.toString() ?? ''));
+
+    return SentryRequest(
+      method: options.method,
+      headers: _options.sendDefaultPii ? headers : null,
+      url: urlWithoutQuery,
+      queryString: query,
+      cookies: _options.sendDefaultPii
+          ? options.headers['Cookie']?.toString()
+          : null,
+      data: _getRequestData(dioError.response?.data),
```

Comment:
You're using request props configuration for the `response` object, which is wrong.
Looks like `DioError` does not have access to the `request` data, but only `response`.

Reply:
Btw adding response data requires discussion yet, See https://github.com/getsentry/sentry-dart/issues/624#issuecomment-1012296140 due to PII concerns, etc...

Reply:
Access to the request options is available.

Reply:
Oh looks like it's a different name, `error.requestOptions.data` and friends would work, indeed.